mesh/mesh.cpp

```diff
@@ -1624,6 +1624,37 @@ void Mesh::SetAttributes()
    }
 }
 
+void Mesh::AttrToMarker(int max_attr, const Array<int> &attrs,
+                        Array<int> &marker) const
+{
+   MFEM_ASSERT(attrs.Max() <= max_attr, "Invalid attribute number present.");
+
+   marker.SetSize(max_attr);
+   if (attrs.Size() == 1 && attrs[0] == -1)
+   {
+      marker = 1;
+   }
+   else
+   {
+      marker = 0;
+      for (auto const &attr : attrs)
+      {
+         MFEM_VERIFY(attr > 0, "Attribute number less than one!");
+         marker[attr-1] = 1;
+      }
+   }
+}
+
+void Mesh::AttrToMarker(const Array<int> &attrs, Array<int> &marker) const
+{
+   AttrToMarker(attributes.Max(), attrs, marker);
+}
```

Comment:
This function should probably still work if both `Mesh::attributes` and `attrs` are empty (in which case `marker` will just be an empty array). But currently it will fail on the call to `attributes.Max()` (and same for `BdrAttrToMarker`).

Reply:
Well isn't that annoying...

I don't believe `Mesh::attributes` can be empty in a valid mesh. On the other hand `attrs` certainly could be and we should handle that case properly. Good catch!

Reply:
Ah, you're right about `attributes`, but `Mesh::bdr_attributes` can be empty (e.g. periodic meshes)

Reply:
Good point. I'll resolve that.